circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/Replica.java

```diff
@@ -302,6 +304,16 @@ public void updateMetadata(
     LOG.info("Updating replica table metadata.");
     TableAndStatistics replicaTable = tableFactory
         .newReplicaTable(eventId, sourceTable, replicaDatabaseName, replicaTableName, tableLocation, replicationMode);
+
+    if (replicationMode == FULL_OVERWRITE) {
+      LOG.debug("Replication mode: FULL_OVERWRITE. Dropping existing replica table and its data.");
+      DropTableService dropTableService = new DropTableService();
+      try {
+        dropTableService.removeTableParamsAndDrop(client, replicaDatabaseName, replicaTableName);
+      } catch (TException e) {
+        LOG.info("No replica table '" + replicaDatabaseName + "." + replicaTableName + "' found. Nothing to delete.");
```

Comment:
I think the `TException` could be thrown for other reasons too, like network problems etc. So perhaps it would be better to have a try/catch for `TException` that retrhrows a `MetaStoreClientException` like in some of the blocks below, and then introduce a "does table exist" check in the code (maybe in the service?) and if it doesn't then it just silently continues without throwing an exception (as this is fine, there's no need to drop the replica if it doesn't exist).

Reply:
NoSuchObjectException is the one that get's throw if the table/db doesn't exist

Reply:
OK, so we could have a more specific catch block for that here, or do we want to change the service to not throw that and silently continue? Do we have other cases that use it where we want that exception thrown?

Reply:
Yeah I also realised this. I have fixed it in the other branch I have for deleting data so its now:
```
} catch (Exception e) {
   LOG.info("Replica table '" + replicaDatabaseName + "." + replicaTableName + "' was not dropped.");
}
```

And in the drop table service it catches the NoSuchObj exception and logs that the table wasnt found.

Reply:
Do you want to update this branch too? The plan is to merge the branches in separately right? First this one to get the "drop metadata" changes in and then the other one gets the "drop file data" changes.

Reply:
Yeah thats the plan. Will update this one with those logs 👍

Reply:
I think it is ok that the DropTableService swallows the NoSuchObjectE. But then I agree with Adrian this block should catch and rethrow MetaStoreClientException. Don't log anything that's not up to this class to silently ignore this.

Reply:
Thats a good point. Will log the NoSuchObj inside the dropTableService, and here will throw a MetaStoreClientExc in the catch.